ppa/archive/management/commands/index_pages.py

```diff
@@ -100,6 +105,55 @@ def handle(self, *args, **kwargs):
             digiworks = DigitizedWork.objects.filter(source_id__in=source_ids)
             num_pages = sum([dw.page_count for dw in digiworks])
 
+        # if reindexing everything, check db totals against solr
+        if not source_ids and self.verbosity >= self.v_normal:
+            # check totals
+            solr_count = self.get_solr_totals()
+
+            work_diff = digiworks.count() - solr_count.get("work", 0)
+            page_diff = num_pages - solr_count.get("page", 0)
```

Comment:
Is this `digiworks.count()` (the total pages in psql db?) guaranteed to be the sum of the individual work page counts that are checked?

Reply:
`digiworks.count()` is the number of digitized works in the db that we expect to be indexed (i.e. public/non-suppressed)

`num_pages` when calculated from `Page.total_to_index` is a sum of all the page counts in the db for published works. When I was testing locally I saw some discrepancies between these numbers, and I need to review the code to look at how that page total is calculated and why it could be different than what's indexed in Solr. (It seems like it must be a bug if they don't match, which is entirely possible.)

I think even if the page counts don't match exactly, this option still gives us a shortcut to substantially decrease the number of pages we index and is still useful even without/before resolving any page count discrepancies between solr and db.